bin/curriculum/migrate_reference_guides.rb

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env ruby
+
+require_relative '../../deployment'
+raise unless [:development, :adhoc, :levelbuilder].include? rack_env
+
+# this script will, for a given course version, update all instances of old format links to concept maps to
+# reference guide links associated with that course version.
+# Links looking like /docs/concepts/patterns/update-screen-pattern/
+# will change to /courses/csp-2022/guides/update-screen-pattern/
+# This will affect the following places:
+#  Help and Tips tab: 'map_reference' and 'reference_links' fields
+#  Map levels: 'reference' field in properties
+
+$verbose = false
+def log(str)
+  puts str if $verbose
+end
+
+$quiet = false
+def warn(str)
+  puts str unless $quiet && !$verbose
+end
+
+# Wait until after initial error checking before loading the rails environment.
+def require_rails_env
+  log "loading rails environment..."
+  start_time = Time.now
+  require_relative '../../dashboard/config/environment'
+  log "rails environment loaded in #{(Time.now - start_time).to_i} seconds."
+end
+
+require_rails_env
+
+# course_name: 'csp-2022'
+def migrate_course_reference_guides(course_name)
+  course_version_id = UnitGroup.find_by_name(course_name).course_version.id
```

Comment:
Since this is a one off script its not a big deal but you could Do the `UnitGroup.find_by_name` once at the top to be used for finding the course_version and the levels.

Reply:
ah of course!